client/components/Item/RelatedItems.vue

```diff
@@ -89,39 +89,56 @@ export default Vue.extend({
       loading: true
     };
   },
-  watch: {
-    item(): void {
-      this.refreshItems();
-    }
+  computed: {
+    ...mapGetters('items', ['getItems', 'getMissingIds'])
   },
-  beforeMount() {
-    try {
-      this.refreshItems();
-    } catch (error) {
-      this.pushSnackbarMessage({
-        message: this.$t('unableGetRelated'),
-        color: 'error'
-      });
+  watch: {
+    'item.Id': {
+      immediate: true,
+      async handler(): Promise<void> {
+        if (this.item.Id) {
+          await this.refreshItems();
+        }
+      }
     }
   },
   methods: {
     ...mapActions('snackbar', ['pushSnackbarMessage']),
     async refreshItems(): Promise<void> {
-      this.loading = true;
+      try {
+        this.loading = true;
 
-      if (this.item.Id) {
-        const response = await this.$api.library.getSimilarItems({
-          itemId: this.item.Id,
-          userId: this.$auth.user?.Id,
-          limit: this.vertical ? 5 : 12
-        });
+        if (this.item.Id) {
```

Comment:
Quick question, what happens when there's no item ID? Is it caught by the trycatch? :)

Reply:
Nothing of the function runs and the function is called again in the watcher.

I noticed though that I had this both in this function and in the watcher, I'll remove it in the watcher to have all the proper type checking.